common/src/main/java/org/apache/celeborn/common/network/buffer/ManagedBuffer.java

```diff
@@ -71,4 +71,15 @@ public abstract class ManagedBuffer {
    * the caller will be responsible for releasing this new reference.
    */
   public abstract Object convertToNetty() throws IOException;
+
+  /**
+   * Convert the buffer into a Netty object, used to write the data out with SSL encryption, which
+   * cannot use {@link io.netty.channel.FileRegion}. The return value is either a {@link
+   * io.netty.buffer.ByteBuf}, a {@link io.netty.handler.stream.ChunkedStream}, or a {@link
+   * java.io.InputStream}.
+   *
+   * <p>If this method returns a ByteBuf, then that buffer's reference count will be incremented and
+   * the caller will be responsible for releasing this new reference.
+   */
+  public abstract Object convertToNettyForSsl() throws IOException;
```

Comment:
The encryption process will happen here in another PR?

Reply:
Yes, that is in a subsequent PR, which will use this method.
You can see this getting used in the consolidated PR (I am afk, so unable to point to exact details unfortunately).

This PR is only setting up the base infra for it.